### ISSUE TITLE
Create a new repository for code relating to the conversion of tensorflow, keras and pytorch models to and from mlpack

### ISSUE THREAD

**Issue by sreenikSS** [mlpack/mlpack#1723]
This is also in accordance with a GSOC idea for the same. Committing the entire work at once is not suitable due to multiple people working on different aspects of the issue, so setting up a new repo would allow for code sharing keeping the original repo intact.

**Comment by sreenikSS** [comment]
Sorry I couldn't label it.
I am working on this issue currently (with tensorflow). I have a doubt which would be better if clarified:

- I am converting tensorflow models to onnx models which would be converted to mlpack models. So will I proceed keeping a dependency on onnx or separately maintain a code similar to that of onnx?

I will keep this thread updated pertaining to any more doubts regarding the process.

**Comment by rcurtin** [comment]
I'd prefer not to create an mlpack repository for this until the project is more mature, but you can definitely create your own repository and the work can be done there.

For TensorFlow and ONNX, I don't have a problem whichever way you want to do it.  If we have a "translator" it's going to necessarily depend on both mlpack and whatever language we are translating to, so most likely we are not going to be able to avoid a dependency on ONNX.

**Comment by bhavya01** [comment]
I had some thoughts regarding this project. In ONNX, computation graphs are made up of a DAG of nodes and each DAG denotes an operation. To convert from onnx models to mlpack models, we can iterate over the nodes in the ONNX model graph, store the layers and params in an ordered dictionary and as suggested here -> https://github.com/mlpack/mlpack/issues/1254#issuecomment-473599654, we can then use this dictionary to create a model in mlpack.

 For this we might need to map every operation in ONNX to the corresponding operation in mlpack. Also, python support for ANN module will be very useful for this project and can be integrated as a part of it.

**Comment by rcurtin** [comment]
@bhavya01 agreed, I think that this is a reasonable strategy that you've outlined here.
